app/controllers/ops_controller/settings/common.rb

```diff
@@ -387,65 +387,65 @@ def settings_update_save
 
       w = qwb[:ems_refresh_worker][:defaults]
-      @edit[:new].set_worker_setting!(:MiqEmsRefreshWorker, %i(defaults memory_threshold), w[:memory_threshold])
+      set_worker_setting(@edit[:new], MiqEmsRefreshWorker, :defaults, :memory_threshold, w[:memory_threshold])
 
-      wb = @edit[:new].config[:workers][:worker_base]
+      wb = @edit[:new][:workers][:worker_base]
       w = wb[:event_catcher]
-      @edit[:new].set_worker_setting!(:MiqEventCatcher, :memory_threshold, w[:memory_threshold])
+      set_worker_setting(@edit[:new], MiqEventCatcher, :memory_threshold, w[:memory_threshold])
 
       w = wb[:vim_broker_worker]
-      @edit[:new].set_worker_setting!(:MiqVimBrokerWorker, :memory_threshold, w[:memory_threshold])
+      set_worker_setting(@edit[:new], MiqVimBrokerWorker, :memory_threshold, w[:memory_threshold])
 
       w = qwb[:smart_proxy_worker]
-      @edit[:new].set_worker_setting!(:MiqSmartProxyWorker, :count, w[:count].to_i)
-      @edit[:new].set_worker_setting!(:MiqSmartProxyWorker, :memory_threshold, w[:memory_threshold])
+      set_worker_setting(@edit[:new], MiqSmartProxyWorker, :count, w[:count].to_i)
+      set_worker_setting(@edit[:new], MiqSmartProxyWorker, :memory_threshold, w[:memory_threshold])
 
       w = wb[:ui_worker]
-      @edit[:new].set_worker_setting!(:MiqUiWorker, :count, w[:count].to_i)
+      set_worker_setting(@edit[:new], MiqUiWorker, :count, w[:count].to_i)
 
       w = qwb[:reporting_worker]
-      @edit[:new].set_worker_setting!(:MiqReportingWorker, :count, w[:count].to_i)
-      @edit[:new].set_worker_setting!(:MiqReportingWorker, :memory_threshold, w[:memory_threshold])
+      set_worker_setting(@edit[:new], MiqReportingWorker, :count, w[:count].to_i)
+      set_worker_setting(@edit[:new], MiqReportingWorker, :memory_threshold, w[:memory_threshold])
 
       w = wb[:web_service_worker]
-      @edit[:new].set_worker_setting!(:MiqWebServiceWorker, :count, w[:count].to_i)
-      @edit[:new].set_worker_setting!(:MiqWebServiceWorker, :memory_threshold, w[:memory_threshold])
+      set_worker_setting(@edit[:new], MiqWebServiceWorker, :count, w[:count].to_i)
+      set_worker_setting(@edit[:new], MiqWebServiceWorker, :memory_threshold, w[:memory_threshold])
 
       w = wb[:websocket_worker]
-      @edit[:new].set_worker_setting!(:MiqWebsocketWorker, :count, w[:count].to_i)
+      set_worker_setting(@edit[:new], MiqWebsocketWorker, :count, w[:count].to_i)
 
-      @update = MiqServer.find(@sb[:selected_server_id]).get_config
+      @update = MiqServer.find(@sb[:selected_server_id]).settings
     when "settings_custom_logos"                                      # Custom Logo tab
-      @changed = (@edit[:new] != @edit[:current].config)
-      @update = VMDB::Config.new("vmdb")                    # Get the settings object to update it
+      @changed = (@edit[:new] != @edit[:current])
+      @update = MiqServer.my_server.settings
```

Comment:
I think I would prefer to use `::Settings.to_hash` anywhere we're doing `MiqServer.my_server.settings`.

I understand if you're looking up a server object and it may or may not be local `#settings` will be better, but if it's definitely local this is more straight-forward in my opinion.

Reply:
It never occurred to me that `MiqServer.my_server.is_local?` is always `true`. I will update this as you suggested.